src/js/components/Chart/Chart.js

```diff
@@ -111,125 +116,138 @@ const renderArea = (
-    } else if (type === 'area') {
-      contents = renderArea(values, bounds, scale, height, this.props);
-    }
+  useEffect(() => {
+    window.addEventListener('resize', onResize);
+    onResize();
 
-    return (
-      <StyledChart
-        ref={this.containerRef}
-        viewBox={viewBox}
-        preserveAspectRatio="none"
-        width={size === 'full' ? '100%' : width}
-        height={size === 'full' ? '100%' : height}
-        {...rest}
-      >
-        <g
-          stroke={normalizeColor(colorName, theme)}
-          strokeWidth={strokeWidth}
-          strokeLinecap={round ? 'round' : 'butt'}
-          strokeLinejoin={round ? 'round' : 'miter'}
-          opacity={opacity}
-        >
-          {contents}
-        </g>
-      </StyledChart>
-    );
+    return function cleanup() {
+      window.removeEventListener('resize', onResize);
+    };
+  }, []);
+
+  if (!boundsState || !valuesState) return null;
+
+  delete rest.values;
+
+  const sizeWidth = typeof size === 'string' ? size : size.width || 'medium';
+  const sizeHeight = typeof size === 'string' ? size : size.height || 'medium';
+  const width =
+    sizeWidth === 'full'
+      ? containerWidth
+      : parseMetricToNum(theme.global.size[sizeWidth] || sizeWidth);
+  const height =
+    sizeHeight === 'full'
+      ? containerHeight
+      : parseMetricToNum(theme.global.size[sizeHeight] || sizeHeight);
+  const strokeWidth = parseMetricToNum(theme.global.edgeSize[thickness]);
+  const scale = [
+    width / (boundsState[0][1] - boundsState[0][0]),
+    height / (boundsState[1][1] - boundsState[1][0]),
+  ];
+  const viewBox = overflow
+    ? `0 0 ${width} ${height}`
+    : `-${strokeWidth / 2} -${strokeWidth / 2} ${width + strokeWidth} ${height +
+        strokeWidth}`;
+  const colorName = typeof color === 'object' ? color.color : color;
+  const opacity = color.opacity
+    ? theme.global.opacity[color.opacity]
+    : undefined;
+
+  let contents;
+  if (type === 'bar') {
+    contents = renderBars(valuesState, boundsState, scale, height);
+  } else if (type === 'line') {
+    contents = renderLine(valuesState, boundsState, scale, height, props);
+  } else if (type === 'area') {
+    contents = renderArea(valuesState, boundsState, scale, height, props);
   }
-}
+
+  return (
+    <StyledChart
+      ref={containerRef}
+      viewBox={viewBox}
+      preserveAspectRatio="none"
+      width={size === 'full' ? '100%' : width}
+      height={size === 'full' ? '100%' : height}
+      {...rest}
+    >
+      <g
+        stroke={normalizeColor(colorName, theme)}
+        strokeWidth={strokeWidth}
+        strokeLinecap={round ? 'round' : 'butt'}
+        strokeLinejoin={round ? 'round' : 'miter'}
+        opacity={opacity}
+      >
+        {contents}
+      </g>
+    </StyledChart>
+  );
+};
+
+Chart.defaultProps = {
+  color: 'accent-1',
+  overflow: false,
+  size: { width: 'medium', height: 'small' },
+  thickness: 'medium',
+  type: 'bar',
+};
```

Comment:
Can we initialize the variables part of deconstructing the props? so we will be more consistent with the behavior of other components.